### PR TITLE
feat(mautic): new contact,contcat updated, contact channel/company  subscription change triggers

### PR DESCRIPTION
**Pull request by kanarelo**
## What does this PR do?

Adds triggers to Mautic piece

commisioned by https://www.platformtraining.com/

ANNOUNCEMENT=true

### CODE REVIEW

File: packages/pieces/community/mautic/package.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "name": "@activepieces/piece-mautic",
-  "version": "0.3.1"
+  "version": "0.3.2"
```
Comment (kishanprmr) [inline packages/pieces/community/mautic/package.json]: please increase the minor version as it introduces the webhook feature.

File: packages/pieces/community/mautic/src/lib/triggers/index.ts [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+export const triggers = [
+  {
+    name: "lead_company_change",
+    displayName: "Contact Updated",
+    description: "Triggered when a contact is updated.",
+    sampleData: {
+      "mautic.lead_company_change": [contactTestData]
+    },
+    eventType: "mautic.lead_company_change",
```
Comment (kishanprmr) [inline packages/pieces/community/mautic/src/lib/triggers/index.ts]: can you please revalidate eventType for the contact updated trigger? 

docs : https://developer.mautic.org/#list-available-webhook-triggers

File: packages/pieces/community/mautic/src/lib/triggers/index.ts [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+    eventType: "mautic.lead_post_save_update",
+  },
+  {
+    name: "lead_channel_subscription_changed",
+    displayName: "Contact Channel Subscription Change",
+    description: "Triggered when a contact's channel subscription status changes.",
+    sampleData: {
+      "mautic.lead_channel_subscription_changed": [contactTestData]
+    },
+    eventType: "mautic.lead_channel_subscription_changed",
```
Comment (kishanprmr) [inline packages/pieces/community/mautic/src/lib/triggers/index.ts:331]: can you please give a reference for this eventType? I couldn't find it in REST API docs.

Comment (kanarelo) [reply packages/pieces/community/mautic/src/lib/triggers/index.ts:331]: You can get this by making a request to "<api>/hooks/triggers". The client funding, is interested in it, Its available on make.com where they are coming from.

Comment (kanarelo) [reply packages/pieces/community/mautic/src/lib/triggers/index.ts:331]: Most likely hidden from documents, dont know why

File: packages/pieces/community/mautic/src/lib/triggers/index.ts (hunk 2 above)
Comment (kanarelo) [reply packages/pieces/community/mautic/src/lib/triggers/index.ts]: Sure

File: packages/pieces/community/mautic/src/lib/triggers/index.ts (hunk 3 above)
Comment (kanarelo) [reply packages/pieces/community/mautic/src/lib/triggers/index.ts:331]: Here is an example of the response from /hooks/triggers
```json
{
  "status": 200,
  "headers": {
    "server": "nginx",
    "date": "Thu, 25 Jan 2024 08:19:53 GMT",
    "content-type": "application/json",
    "transfer-encoding": "chunked",
    "connection": "close",
    "vary": "Accept-Encoding, Authorization",
    "mautic-version": "5.0.2",
    "cache-control": "no-cache, private"
  },
  "body": {
    "triggers": {
      "mautic.company_post_save": {
        "label": "Company Create/Update Event",
        "description": "Triggered when a company is created/updated"
      },
      "mautic.company_post_delete": {
        "label": "Company Deleted Event",
        "description": "Triggered when a company is deleted"
      },
      "mautic.lead_channel_subscription_changed": {
        "label": "Contact Channel Subscription Change Event",
        "description": "Triggered when a contact's channel subscription status changes."
      },
      "mautic.lead_company_change": {
        "label": "Contact Company Subscription Change Event",
        "description": "Triggered when a commpany is added or removed to/from contact"
      },
      "mautic.lead_post_delete": {
        "label": "Contact Deleted Event",
        "description": "Triggered when a contact is deleted."
      },
      "mautic.lead_post_save_new": {
        "label": "Contact Identified Event",
        "description": "Triggered when a contact is identified."
      },
      "mautic.lead_points_change": {
        "label": "Contact Points Changed Event",
        "description": "Triggered when a contact's points are modified."
      },
      "mautic.lead_list_change": {
        "label": "Contact Segment Membership Change Event",
        "description": "Triggered when a contact segment membership is changed"
      },
      "mautic.lead_post_save_update": {
        "label": "Contact Updated Event",
        "description": "Triggered when a contact is updated."
      },
      "mautic.email_on_open": {
        "label": "Email Open Event",
        "description": "mautic.email.webhook.event.open_desc"
      },
      "mautic.email_on_send": {
        "label": "Email Send Event",
        "description": "mautic.email.webhook.event.send_desc"
      },
      "mautic.form_on_submit": {
        "label": "Form Submit Event",
        "description": "mautic.form.webhook.event.form.submit_desc"
      },
      "mautic.page_on_hit": {
        "label": "Page Hit Event",
        "description": "mautic.page.webhook.event.hit_desc"
      },
      "mautic.sms_on_send": {
        "label": "Text Send Event",
        "description": "mautic.sms.webhook.event.send_desc"
      }
    }
  }
}
```

Comment (kishanprmr) [reply packages/pieces/community/mautic/src/lib/triggers/index.ts:331]: @kanarelo Thanks for the details.

**Review by kishanprmr — APPROVED** [review]

**Pull request merged**